### PR TITLE
@damassi => [WIP] [FilterArtworks] Save a redundant call happening during filter usage

### PR DESCRIPTION
**Pull request by mzikherman**
Will still add some specs.

This removes the second call, from the screenshot below.

Screen Shot 2019-07-23 at 1 16 51 PM

Basically, we try to skip this fetch if we're requesting fields that either don't need it, or resolve/fetch their own data.

This was already the intent here, and worked when you used string interpolation to send one giant query, but with sending Relay-style fragments and spreads, would fail (the AST check is naïve). This uses @ashkan18 's excellent work to make this more sophisticated.

WIP since I'll add an explicit spec (although we have specs asserting that a fetch is made/does not happen based on what is queried, so those still passing with these changes is a good sign).

### CODE REVIEW

File: src/schema/v1/filter_artworks.ts [hunk 1]
```diff
@@ -384,8 +381,20 @@ const filterArtworksTypeFactory = (
...
+      "artworks_connection",
+      "__id",
+      "edges", // comes from `artworks_connection`
+    ]
+    const requireFetch = includesFieldsOtherThanSelectionSet(
+      info,
+      fieldsNotRequiringFetching
+    )
+
+    console.log(requireFetch)
```
Comment (damassi) [inline src/schema/v1/filter_artworks.ts]: stray loggy

**Comment by joeyAghion** [comment]
For reference: https://artsyproduct.atlassian.net/browse/PLATFORM-1622

**Comment by ashkan18** [comment]
🙌

**Comment by mzikherman** [comment]
I'll keep this open, but superseded by https://github.com/artsy/metaphysics/pull/1845.

**Comment by mzikherman** [comment]
Actually, I'll close this one for now.

**Pull request closed without merging**